source/server-discovery-and-monitoring/tests/rs/primary_mismatched_me_not_removed.yml

```diff
@@ -0,0 +1,69 @@
+description: Primary mismatched me is not removed
+uri: mongodb://localhost:27017,localhost:27018/?replicaSet=rs
+
+phases: [
+  {
+    responses: [
+      ["localhost:27017", {
+        ok: 1,
+        hosts: [
+          "localhost:27017",
+          "localhost:27018"
+        ],
+        ismaster: true,
+        setName: "rs",
+        primary: "localhost:27017",
+        # me does not match the primary responder's address, but the server
+        # is still added because we don't me mismatch check the primary and all
+        # servers from a primary ismaster are added to the working server set
+        me: "a:27017",
+        minWireVersion: 0,
+        maxWireVersion: 7
+      }]
+    ],
+    outcome: {
+      servers: {
+        "localhost:27017": {
+          type: "RSPrimary",
+          setName: "rs"
+        },
+        "localhost:27018": {
+          type: "Unknown",
+          setName: null
+        }
+      },
+      topologyType: "ReplicaSetWithPrimary",
+      logicalSessionTimeoutMinutes: null,
+      setName: "rs"
+    }
+  },
+  {
+    responses: [
```

Comment:
Nit: this is proving that secondary mismatched me is removed, which is not what the description of the test says.  Don't we already test this somewhere else?

Reply:
It does prove that, but I don't think you can write this test without two nodes. Do you feel strongly about this?

Reply:
I'm not objecting to the test.  Just that it's included in a test with the description: `Primary mismatched me is not removed`.  I think it should be a separate test with the description: `Secondary mismatched me is removed`.

My separate question was whether there's already a test for this elsewhere.  Is this testing something different than the `Secondary mismatched me` test that we already have?